CMakeLists.txt

```diff
@@ -47,6 +47,8 @@ if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
   project(stablehlo LANGUAGES CXX C)
   set(CMAKE_C_STANDARD 11)
   set(CMAKE_CXX_STANDARD 17)
+  set(LLVM_ENABLE_WARNINGS ON)
+  set(LLVM_ENABLE_WERROR ON)
```

Comment:
I have very strong concerns with this. 

Werror should never be on by default: this can make it fail on someone's machine just because they are using a different compiler, C++ library, or even C library!

My understanding from the issue was that you were looking to turn this on in CI, which is a very different value proposition (you control the environment entirely).

Reply:
That sounds reasonable. @burmako any opposition to removing this from the CMakeLists and adding these as flags in `ci_build_stablehlo.sh` so we only build with warnings as errors in CI? We can still verify our changes locally with ease by calling the build script.

Reply:
> We can still verify our changes locally with ease by calling the build script.

You should provide a docker image for this, otherwise you're still dependent on the environment (compiler version and libraries on the host)